src/main/java/org/llamarama/petty/mixins/ParrotEntityMixin.java

```diff
@@ -0,0 +1,59 @@
+package org.llamarama.petty.mixins;
+
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.Flutterer;
+import net.minecraft.entity.passive.ParrotEntity;
+import net.minecraft.entity.passive.TameableShoulderEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.ItemScatterer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.llamarama.petty.blocks.CageBlock;
+import org.llamarama.petty.blocks.ModIdBlocks;
+import org.llamarama.petty.items.ModIdItems;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ParrotEntity.class)
+public abstract class ParrotEntityMixin extends TameableShoulderEntity implements Flutterer {
+    private int featherDrop = 10;
+
+    public ParrotEntityMixin(EntityType<? extends ParrotEntity> entityType, World world) {
+        super(entityType, world);
+    }
+
+    @Inject(at = @At("HEAD"), method = "interactMob", cancellable = true)
+    public void interactMob(PlayerEntity player, Hand hand, CallbackInfoReturnable<ActionResult> cir) {
+        BlockPos cagePos = this.getBlockPos();
+        if (player.getStackInHand(hand).isOf(ModIdBlocks.INSTANCE.getBIRD_CAGE().asItem())) {
+            if (this.isSitting() || this.isInSittingPose()) {
+                if (player.getStackInHand(hand).hasCustomName()) this.setCustomName(player.getStackInHand(hand).getName());
```

Comment:
Does this override a parrots name? Because you might have renamed the parrot before putting it in the cage... I think we should check that?

Reply:
Well if you've purposely named the bird cage, then you probably want to change the parrot's name. I don't see why you would name it otherwise.